misago/users/models/user.py

```diff
@@ -91,6 +93,14 @@ def get_by_username_or_email(self, login):
             return self.get(email_hash=hash_email(login))
         return self.get(slug=slugify(login))
 
+    def prepare_new_username(self, username):
```

Comment:
This function should live in utility module and not in manager.

Reply:
I don't want to import User model in utils.py file

Reply:
This util should live in `misago.sso` IMHO so its not imported in `users.utils` causing circular dependency.